pkg/services/cloudmigration/gmsclient/gms_client.go

```diff
@@ -148,8 +158,54 @@ func (c *gmsClientImpl) StartSnapshot(ctx context.Context, session cloudmigratio
 	return &result, nil
 }
 
-func (c *gmsClientImpl) GetSnapshotStatus(context.Context, cloudmigration.CloudMigrationSession, cloudmigration.CloudMigrationSnapshot) (*cloudmigration.CloudMigrationSnapshot, error) {
-	panic("not implemented")
+func (c *gmsClientImpl) GetSnapshotStatus(ctx context.Context, session cloudmigration.CloudMigrationSession, snapshot cloudmigration.CloudMigrationSnapshot) (*cloudmigration.GetSnapshotStatusResponse, error) {
+	c.getStatusMux.Lock()
+	defer c.getStatusMux.Unlock()
+	logger := c.log.FromContext(ctx)
+
+	// Ensure we can't send requests more frequently than supported
+	if !c.getStatusLastQueried.IsZero() && time.Since(c.getStatusLastQueried) < c.minGetStatusPollingPeriod {
+		logger.Debug("skipping query to Grafana Migration Service because this request was made too soon after the previous one")
+		return &cloudmigration.GetSnapshotStatusResponse{
+			State: cloudmigration.SnapshotStateUnknown,
+		}, nil
+	}
+
+	path := fmt.Sprintf("%s/api/v1/status/%s/status", buildBasePath(c.domain, session.ClusterSlug), snapshot.GMSSnapshotUID)
+
+	// Send the request to gms with the associated auth token
+	req, err := http.NewRequest(http.MethodGet, path, nil)
+	if err != nil {
+		c.log.Error("error creating http request to get snapshot status", "err", err.Error())
+		return nil, fmt.Errorf("http request error: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %d:%s", session.StackID, session.AuthToken))
+
+	client := &http.Client{}
```

Comment:
not urgent, but this could be created once and not for every requrest

Reply:
Going to leave as-is for now, we do have different timeouts per operation which I am now passing into the http client. Maybe as a future refactor we can make this all a one liner for each method.

Reply:
:+1: 
might be worth adding at TODO comment. We could have clients for each operation if they have different timeouts.

Reply:
https://github.com/grafana/grafana/pull/90314/commits/6f69f5b5c6c372cc7a54d6e7f68ca23ba0b2cdb1